modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Utils.java

```diff
@@ -55,6 +57,19 @@ public static void setup() {
 
     }
 
+    private static AtomicBoolean isAvailableProcessorsSet = new AtomicBoolean();
+
+    /**
+     * Set the number of available processors that Netty uses for sizing various resources (e.g., thread pools).
```

Comment:
@nik9000 Are you okay with an assert as in eccfc39dd8bc0b3712bb383621c127f32a0e684e or would you prefer a hard exception?